packages/core/lib/match.js

```diff
@@ -15,18 +15,21 @@ module.exports = match
  * @param {Function[]} configSetters            Array of functions as returned by webpack blocks.
  * @return {Function}
  */
-function match (test, options, configSetters) {
+function match (patterns, options, configSetters) {
   if (!configSetters && Array.isArray(options)) {
     configSetters = options
     options = {}
   }
 
   assertConfigSetters(configSetters)
 
-  const match = { test: createFileTypeMatcher(test) }
+  const match = {}
 
-  if (options.exclude) {
-    match.exclude = options.exclude
+  const [test, negations] = splitPatterns(patterns)
```

Comment:
I would rather put the `typeof patterns === 'string'` here. Would make the splitPatterns() function much simpler.

And I think there are basically just two use cases:
1. `patterns: string[]`
2. `patterns: string`

Reply:
Maybe we should also still call it `test`, since `patterns` sounds like `string[]`.

👉 `splitPatterns(typeof test === 'string' ? [test] : test)`